### ISSUE TITLE
windows_package broken in 12.7.2

### ISSUE THREAD

**Issue by TemporarilyOffline** [chef/chef#4577]
It appears that the resource is not downloading the file.  Subsequently when the install action is attempted, there is nothing to install.

2 Examples:

```           ================================================================================
           Error executing action `install` on resource 'windows_package[Ruby 2.2.3-p173-x64]'
           ================================================================================

           Errno::ENOENT
           -------------
           No such file or directory - C:/Users/ADMINI~1/AppData/Local/Temp/kitchen/cache/package/rubyinstaller-2.2.3-x64.exe

           Resource Declaration:
           ---------------------
           # In C:/Users/ADMINI~1/AppData/Local/Temp/kitchen/cache/cookbooks/cookboook/recipes/ruby.rb

             1: windows_package 'Ruby 2.2.3-p173-x64' do
             2:   source node['ruby']['source']
             3:   checksum node['ruby']['checksum']
             4:   action :install
             5: end
```

```           ================================================================================
           Error executing action `install` on resource 'windows_package[.NET 4.5.2 Developer Pack]'
           ================================================================================

           Errno::ENOENT
           -------------
           No such file or directory - C:/Users/ADMINI~1/AppData/Local/Temp/kitchen/cache/package/NDP452-KB2901951-x86-x64-DevPack.exe

           Resource Declaration:
           ---------------------
           # In C:/Users/ADMINI~1/AppData/Local/Temp/kitchen/cache/cookbooks/cookbook_name/recipes/dot_net_4.5.2_developer_pack.rb

             1: windows_package '.NET 4.5.2 Developer Pack' do
             2:   source node['dot_net_dev_pack']['source']
             3:   checksum node['dot_net_dev_pack']['checksum']
             4:   options '/norestart /q'
             5:   action :install
             6: end
```

Running this same code in 12.6.0 returns: 

```
       Recipe: cookbook_name::dot_net_4.5.2_developer_pack
         * windows_package[.NET 4.5.2 Developer Pack] action install
```

Thanks!

**Comment by jaym** [comment]
The problem is we're trying to check what provider to use before we have to file to warn the user in case they're doing something that's going to break: https://github.com/chef/chef/blob/master/lib/chef/provider/package/windows.rb#L38-L41

Perhaps we can warn for this case at some later point? cc @mwrock

**Comment by mwrock** [comment]
I saw this testing something else on 12.7 Friday. I'll have a look tomorrow or sooner.

**Comment by spuder** [comment]
Additional Datapoint: 
I have a similar resource in my cookbooks. Concerned that 12.7.2 might break it, I ran some tests. I am unable to reproduce with my usecase. 

    default['nd-web']['wce'] = 'http://download.microsoft.com/download/5/5/1/5511bfc6-e52f-4db0-bafb-fd5dcb91eff0/Microsoft%20WSE%203.0.msi'

```
# OK in 12.7.2
windows_package "WCE" do
  package_name "Microsoft WSE 3.0"
  source node['nd-web']['wce']
  installer_type :msi
  action :install
end
```
The following also work properly

```
# OK in 12.7.2
windows_package "WCE" do
  package_name "Microsoft WSE 3.0"
  source node['nd-web']['wce']
  checksum 'dc6c1ebfc4fe7166d182e24a89f3338c9df18f66526e521cec184a0019d181a1'
  installer_type :msi
  action :install
end
```

```
# OK in 12.7.2
windows_package "WCE" do
  package_name "Microsoft WSE 3.0"
  source node['nd-web']['wce']
  checksum 'dc6c1ebfc4fe7166d182e24a89f3338c9df18f66526e521cec184a0019d181a1'
  installer_type :msi
  options '/silent'
  action :install
end
```

**Comment by TemporarilyOffline** [comment]
Interesting... neither of my packages are MSIs

**Comment by mwrock** [comment]
fixed by #4612
